docs_src/microservices/general/messagebus.md

```diff
@@ -0,0 +1,368 @@
+    
+        All events coming from any device service for any device profile, device or source
+      
+    - **edgex/events/+/+/camera-001/#**
+    
+        Events coming from any device service or core data for any device profile, but only for the device "camera-001" and for any source
+      
+    - **edgex/events/device/onvif/+/status**
+    
+        Events coming from any device service for only the device profile "onvif", and any device and only for the source "status"
+
+
+
+## Deployment
+
+### Redis Pub/Sub (default)
+
+All EdgeX services are capable of using the Redis Pub/Sub without any changes to configuration. The released compose files and snaps use Redis Pub/Sub.
+
+### MQTT 3.1
+
+All EdgeX services are capable of using MQTT 3.1 by simply making changes to each service's configuration. 
+
+!!! note
+    As mentioned above, the MQTT 3.1 implementation requires the addition of a MQTT Broker service to be running.
+
+#### Configuration Changes
+
+!!! example - "Example MQTT Configurations changes for Core/Support/Device Services"
+    The following `MessageQueue` configuration settings must be changed for all EdgeX Core/Support/Device Services to use MQTT 3.1
+    ```toml
+    [MessageQueue]
+    Type = "mqtt"
+    Protocol = "tcp" 
+    Host = "localhost" # in docker this must be overriden to be the docker host name of the MQTT Broker
+    Port = 1883
+    AuthMode = "none"  # Currently in secure mode the MQTT Broker is not secured
+      [MessageQueue.Optional]
+      ClientId ="<service-key>" # must be unique name of the service, thus the service key (core-data, etc) is used
+      Qos          =  "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+      KeepAlive    =  "10" # Seconds (must be 2 or greater)
+      Retained     = "false"
+      AutoReconnect  = "true"
+      ConnectTimeout = "5" # Seconds
+    ```
+
+!!! example - "Example MQTT Configurations changes for Application Services"
+    The following `Trigger.EdgexMessageBus` configuration settings must be changed for all EdgeX Application Services to use MQTT 3.1
+
+    ```toml
+    [Trigger]
+    Type="edgex-messagebus"
+      [Trigger.EdgexMessageBus]
+      Type = "mqtt"
+        [Trigger.EdgexMessageBus.SubscribeHost]
+        Protocol = "tcp"
+        Host = "localhost" # in docker this must be overriden to be the docker host name of the MQTT Broker
+        Port = 1883
+        [Trigger.EdgexMessageBus.PublishHost]
+        Protocol = "tcp"
+        Host = "localhost" # in docker this must be overriden to be the docker host name of the MQTT Broker
+        Port = 1883
+        [Trigger.EdgexMessageBus.Optional]
+        ClientId ="<service-key>" # must be unique name of the service, thus the service key (app-rule-engine, etc) is used
+        Qos          =  "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+        KeepAlive    =  "10" # Seconds (must be 2 or greater)

```

Comment:
"10s" or "10"? the above documentation seems to suggest using 10s

Reply:
Good catch. KeepAlive is also an integer. Above needs to be corrected.